### ISSUE TITLE
"Receive push notifications" and "Allow device attestation" toggles are greyed out

### ISSUE THREAD

**Issue by UnlimitedCookies** [microg/GmsCore#2002]
Both services seem to be broken and FCM shows disconnected state. I cannot toggle them off and microg doesn't give me a proper error message.

No idea how to reproduce, but everything used to work perfectly.

**Screenshots**
![Screenshot_20230801-234847_microG Services Core_1](https://github.com/microg/GmsCore/assets/49200216/6766e627-c394-4f1b-9101-4901b707041e)

**System**
Android Version: 12
Custom ROM: crDroid 8.7

**Comment by UnlimitedCookies** [comment]
Here's what I found in the syslog searching for microg:

[2023-08-02-00-01-23-microg-broken.txt](https://github.com/microg/GmsCore/files/12234351/2023-08-02-00-01-23-microg-broken.txt)

**Comment by UnlimitedCookies** [comment]
Okay, after a restart the connection to FCM could be reestablished, but the toggles remain petrified.

**Comment by mar-v-in** [comment]
From the log excerpt it looks as if device registration is disabled.

**Comment by UnlimitedCookies** [comment]
> From the log excerpt it looks as if device registration is disabled.

Correct, google device registration is disabled. Does that have to be enabled, because both FCM and Safetynet seemed to work with it being turned off.

**Comment by Sapiosenses** [comment]
> > From the log excerpt it looks as if device registration is disabled.
> 
> Correct, google device registration is disabled. Does that have to be enabled, because both FCM and Safetynet seemed to work with it being turned off.



Yes, it says right in the Cloud Messaging settings page that device registration must be enabled for that service to work.

If the device does not register to Google push servers, they do not know where to send the messages from your apps.

**Comment by UnlimitedCookies** [comment]
Then the bug lies in me being able to sometimes use FCM without having my device registered.
1. Why did this even work?
2. Should the FCM settings page enforce the activation of device registration?
